mkosi/distributions/opensuse.py

```diff
@@ -94,6 +111,42 @@ def zypper_modifyrepo(state: MkosiState, repo: str, caching: bool) -> None:
     invoke_zypper(state, [], "modifyrepo", ["--keep-packages" if caching else "--no-keep-packages"], repo)
 
 
+def zypper_init_repositories(state: MkosiState) -> None:
+    # If we need to use a local mirror, create a temporary repository
+    # definition, which is valid only at image build time. It will be removed
+    # from the image and replaced with the final repositories at the end of the
+    # installation process.
+    #
+    # We need to enable packages caching in any cases to make sure that the
+    # package cache stays populated after "zypper install".
+
+    if state.config.local_mirror:
+        zypper_addrepo(state, state.config.local_mirror, "local-mirror", caching=True)
+    else:
+        for (name, url) in OpensuseInstaller.repositories(state):
```

Comment:
Style nit:
```suggestion
        for name, url in OpensuseInstaller.repositories(state):
```
same below